search/services/fulltext.py

```diff
@@ -0,0 +1,100 @@
+"""Provides access to fulltext content for arXiv papers."""
+
+from functools import wraps
+import os
+from urllib.parse import urljoin
+import json
+
+import requests
+
+from arxiv import status
+from search.context import get_application_config, get_application_global
+from search.domain import Fulltext
+
+
+class FulltextSession(object):
+    """An HTTP session with the fulltext endpoint."""
+
+    def __init__(self, endpoint: str) -> None:
+        """Initialize an HTTP session."""
+        self._session = requests.Session()
+        self._adapter = requests.adapters.HTTPAdapter(max_retries=2)
+        self._session.mount('https://', self._adapter)
+
+        if not endpoint[-1] == '/':
+            endpoint += '/'
+        self.endpoint = endpoint
+
+    def retrieve(self, document_id: str) -> Fulltext:
+        """
+        Retrieve fulltext content for an arXiv paper.
+
+        Parameters
+        ----------
+        document_id : str
+            arXiv identifier, including version tag. E.g. ``"1234.56787v3"``.
+        endpoint : str
+            Base URL for fulltext endpoint.
+
+        Returns
+        -------
+        :class:`.Fulltext`
+            Includes the content itself, creation (extraction) date, and
+            extractor version.
+
+        Raises
+        ------
+        ValueError
+            Raised when ``document_id`` is not a valid arXiv paper identifier.
+        IOError
+            Raised when unable to retrieve fulltext content.
+        """
+        if not document_id:    # This could use further elaboration.
```

Comment:
We may have better sanitization checks in javascript, but can I confirm those exist with test coverage?

Reply:
That's a great question. We should address this explicitly: https://github.com/cul-it/arxiv-search/issues/111